mitiq/pt/pt_utils.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (C) Unitary Fund
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+
+"""Utility functions for Pauli Twirling of CZ & CNOT gates."""
+import cirq
+import itertools
+import numpy as np
+
+from mitiq.utils import PAULIS, arbitrary_tensor_product, matrix_to_vector
+
+
+def _get_n_qubit_paulis(num_qubits):
+    """Get list of n-qubit Pauli unitaries."""
+    # get the n-qubit paulis from the Pauli group
+    # disregard the n-qubit paulis with complex phase
+    n_qubit_paulis = PAULIS
+    for i in range(num_qubits - 1):
+        n_qubit_paulis = n_qubit_paulis
+        empty_pauli_list = []
+        for j in PAULIS:
+            for k in n_qubit_paulis:
+                new_pauli = arbitrary_tensor_product(j, k)
+                empty_pauli_list.append(new_pauli)
+        n_qubit_paulis = empty_pauli_list
+    return n_qubit_paulis
+
+
+def _pauli_vectorized_list(num_qubits):
+    """Define a function to create a list of vectorized matrices.
+
+    If the density matrix of interest has more than n>1 qubits, the
+    Pauli group is used to generate n-fold tensor products before
+    vectorizing the unitaries.
+    """
+    n_qubit_paulis = _get_n_qubit_paulis(num_qubits)
+    output_pauli_vec_list = []
+    for i in n_qubit_paulis:
+        vec_of_pauli_matrix = np.reshape(matrix_to_vector(i), (-1, 1))
+        output_pauli_vec_list.append(vec_of_pauli_matrix)
+    return output_pauli_vec_list
+
+
+def pauli_transfer_matrix(circuit):
+    """Find the Pauli Transfer Matrix of an input circuit.
+
+    The circuit can be noiseless or ideal.
+    """
+    num_qubits = len(circuit.all_qubits())
+    try:
+        superop = cirq.kraus_to_superoperator(cirq.kraus(circuit))
+    except TypeError:
+        kraus_list = []
+        for i in range(len(circuit)):
+            kraus_of_op = cirq.kraus(circuit[i])
+            kraus_list.append(kraus_of_op)
+        all_kraus = list(itertools.chain(*kraus_list))
+        superop = cirq.kraus_to_superoperator(all_kraus)
```

Comment:
```suggestion
    except TypeError:
        kraus_list = []
        for i in range(len(circuit)):
            kraus_of_op = cirq.kraus(circuit[i])
            kraus_list.append(kraus_of_op)
        all_kraus = list(itertools.chain(*kraus_list))
        superop = cirq.kraus_to_superoperator(all_kraus)
```
@Misty-W @andreamari Would either of you mind looking at these lines for me? I am trying to get the superoperator of a noisy circuit or a twirled version of the noisy circuit. Unfortunately, there is an error that I can't find. 

The norm of this superoperator is not 4 (as expected) for a two-qubit circuit with a CNOT gate and depolarizing noise, twirled versions of the noiseless circuit or twirled versions of the circuit with depolarized noise. 

I believe this is screwing up my averaged over PTM off-diagonal terms. I do think for the noiseless circuit, the off-diagonal terms should become zero after twirling different versions of the input circuit. 

I know the tests are failing right now because I have not added any tests for the new functions. I also know I have skipped type annotations in the defined functions. Going to leave this for later.